### DISCUSSION TITLE
External sensor NTC

### DISCUSSION THREAD

**Discussion by proddy** [emsesp/EMS-ESP32#2169] [Q&A]
Got a question via the web form:

> Hi, i want to include an Buderus NTC Sensor additionally. Setting the GPIO was OK, but how can I include the calculation from ADC to temperature values (NTC is non linear).

**Comment by MichaelDvP** [comment]
Using a dallas is much easier. For NTC he need a const current source or a fixed reference resistor.
- Measure the adc
- create a custom ram val for the temperature
- create a scheduler on change of adc and calculate the temperature as command for the custom value. We have shortly implemented the log function (natural log)  he needs.
